sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -605,6 +606,9 @@ def n_iter_(self):
         check_is_fitted(self, '_predictors')
         return len(self._predictors)
 
+    def _more_tags(self):
+        return {'allow_nan': True}
```

Comment:
This is not strictly True, but it's needed since `force_all_finite=False` means accept **both** nans and infinite values. Without the tag, common checks would fail.

Reply:
I would leave what you said as a comment in the code.

Reply:
Ok, i'll have to remember to remove it when we merge then